src/main/java/com/iota/iri/service/API.java

```diff
@@ -354,7 +363,7 @@ private boolean isBelowNewTransactionLimit(InetAddress sourceAddress, int size)
             broadcastStoreTimer.set(now);
         }
 
-        if(broadcastStoreCounters.putIfAbsent(sourceAddress, new AtomicInteger(size)) != null) {
+        if(broadcastStoreCounters.putIfAbsent(sourceAddress, new AtomicInteger(size)) == null) { //not already in counter
             return size < newTransactionsLimit;
```

Comment:
this is a potential data race (parallel setApiRateLimit + read)
make this atomic?